src/Features/CSharp/Portable/MakeLocalFunctionStatic/GetCaptures.cs

```diff
@@ -0,0 +1,189 @@
+
+
+
+
+            var parameters = DetermineParameters(captures);  
+
+
+
+
+            //Finds all the call sites of the local function
+            var workspace = document.Project.Solution.Workspace;
+            var arrayNode = await SymbolFinder.FindReferencesAsync(localFunctionSymbol, document.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+
+
+
+            //Initializes a dictionary to replace the nodes of the call sites to be filled with arguments
+            Dictionary<SyntaxNode, SyntaxNode> dict = new Dictionary<SyntaxNode, SyntaxNode>();
+
+
+
+
+            //keep trivia
+            foreach (var referenceSymbol in arrayNode)
+            {
+                foreach (var location in referenceSymbol.Locations)
+                {
+                    var root = await location.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var syntaxNode = root.FindNode(location.Location.SourceSpan); //Node for the identifier syntax
+
+
+
+
+                    var invocation = (syntaxNode as IdentifierNameSyntax).Parent as InvocationExpressionSyntax;
+                    if (invocation == null)
+                    {
+                        return document.Project.Solution;
+                    }
+
+
+
+                    var arg_List = invocation.ArgumentList;
+                    List<ArgumentSyntax> x = new List<ArgumentSyntax>();
+
+
+
+                    foreach (var parameter in parameters)
+                    {
+
+
+
+
+                        var newArgument = GenerateArgument(parameter, parameter.Name, false);
+
+
+
+                        x.Add(newArgument as ArgumentSyntax);
+                    }
+
+
+
+                    var newArgList = arg_List.WithArguments(arg_List.Arguments.AddRange(x));
+                    var newInvocation = invocation.WithArgumentList(newArgList);
+
+
+
+                    dict.Add(invocation, newInvocation);
+                }
+            }
+
+
+
+
+            //Updates the declaration with the variables passed in
+            var newLF = CodeGenerator.AddParameterDeclarations(localfunction, parameters, workspace);
+
+            //Adds the modifier static
+            var modifiers = DeclarationModifiers.From(localFunctionSymbol).WithIsStatic(true);
+            var LFWithStatic = s_generator.WithModifiers(newLF, modifiers);
+
+
+            dict.Add(localfunction, LFWithStatic);
+            var syntaxTree = localfunction.SyntaxTree;
+
+
+
+            var newRoot = syntaxTree.GetRoot(cancellationToken).ReplaceNodes(dict.Keys, (invocation, _) => dict[invocation]);
+            var newDocument = document.WithSyntaxRoot(newRoot);
+
+
+
+            return newDocument.Project.Solution;
+
+
+
+            //Gets all the variables in the local function and its attributes and puts them in an array
+            static ImmutableArray<IParameterSymbol> DetermineParameters(ImmutableArray<ISymbol> captures)
```

Comment:
nit: could we have a more descriptive name for this helper? `DetermineParameters` feels kinda vague to me. A good name should give you an idea what the method does w/o looking at the implementation.